compositor_common/src/scene/text_spec.rs

```diff
@@ -116,3 +122,18 @@ impl From<&TextSpec> for AttrsOwned {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum TextResolution {
+    /// Renders text and "trims" texture to smallest possible size
+    Fitted {
+        /// Must be specified with wrap
+        max_width: Option<u32>,
+        #[serde(default = "default_max_height")]
+        max_height: u32,
+    },
```

Comment:
If you want to keep your current behavior then
```suggestion
    Fitted {
        #[serde(default = "default_max_width")]
        max_width: u32,
        #[serde(default = "default_max_height")]
        max_height: u32,
    },
```
but I don't really like a few things about this approach
- treating our limitations as defaults seems a bit weird
- no way to set the exact width when wrapping

on the other hand, this approach is making an API very simple

so maybe to combine those approaches
```suggestion
    Fitted {
        #[serde(default = "default_max_width")]
        max_width: u32,
        #[serde(default = "default_max_height")]
        max_height: u32,
    },
    ColumnFitted {
        width: u32,
        #[serde(default = "default_max_height")]
        max_height: u32,
    },
```